src/main/kotlin/de/simpletactics/wiki/lib/adapter/persistence/PollSqlAdapter.kt

```diff
@@ -0,0 +1,179 @@
+            IdMapper(),
+            poll.question,
+            poll.description,
+            date,
+            poll.ended,
+            jsonParser.parsetoJson(poll.pollEntries)
+        ).first()
+    }
+
+    fun getPoll(id: Int): PollEntity? {
+        val pollEntry = jdbc.query("select * from wiki_poll where id = ?", PollMapper(), id)
+        return if (pollEntry.isNotEmpty()) {
+            pollEntry.first()
+        } else {
+            null
+        }
+    }
+
+    fun updatePoll(poll: PollEntity): PollEntity? {
+        return if (poll.id != null) {
+            val pollFromDb = getPoll(poll.id)
+
+            if (pollFromDb != null) {
+                updatingEntries(poll, pollFromDb)
+            }
+            getPoll(poll.id)
+        } else {
+            throw WikiNotFoundException("Couldn't update poll. Poll id is null")
+        }
+    }
+
+    fun setPollVotes(
+        userId: Int,
+        pollModel: PollEntity,
+    ) {
+        val oldPoll = getPoll(pollModel.id!!)
+
+        if (oldPoll != null) {
+            pollModel.pollEntries.forEach {
+                val option = getUserVoteOption(it)
+                val date = Date.getDateAsString()
+
+                jdbc.execute(
+                    "UPDATE wiki_poll set data = (" +
+                            "SELECT JSONB_AGG(" +
+                            "CASE when d.value @? '$.pollOption.uuid ? (@ == \"${it.pollOption.uuid}\")' " +
+                            "THEN " +
+                            "CASE when d.value @? '$.votes ?(@.userId == $userId)' " +
+                            "THEN jsonb_set(d.value, '{votes}', " +
+                            "(SELECT JSONB_AGG( " +
+                            "CASE when dd.value @? '$ ?(@.userId == $userId)' AND NOT dd.value -> 'option' @> '\"$option\"'::jsonb " +
+                            "THEN jsonb_set(jsonb_set(dd.value," +
+                            "'{option}', '\"$option\"')" +
+                            ", '{date}', '\"$date\"') " +
+                            "ELSE dd.value " +
+                            "END) " +
+                            "FROM jsonb_array_elements(jsonb_path_query_array(d.value, '$.votes[*]')) dd)" +
+                            " ) " +
+                            "ELSE jsonb_insert(d.value, '{votes, -1}', '{" +
+                            "\"userId\": $userId," +
+                            "\"date\": \"$date\"," +
+                            "\"option\": \"$option\"" +
+                            "}', true) " +
+                            "END " +
+                            "ELSE d.value " +
+                            "END) " +
+                            "FROM jsonb_array_elements(data) d) " +
+                            "where id = ${pollModel.id}",
+                )
+            }
+        } else {
+            log.warn("Could not set poll votes because there is no poll with id ${pollModel.id}")
+        }

```

Comment:
Würde hierzu tendieren:
        if (pollModel.id != null && getPoll(pollModel.id) == null) {
            log.warn("Could not set poll votes because there is no poll with id ${pollModel.id}")
            return
        }

        pollModel.pollEntries.forEach {
            val option = getUserVoteOption(it)
            val date = Date.getDateAsString()

            jdbc.execute(
                """UPDATE wiki_poll set data = (
                        SELECT JSONB_AGG(
                        CASE when d.value @? '$.pollOption.uuid ? (@ == \"${it.pollOption.uuid}\")'
                        THEN
                        CASE when d.value @? '$.votes ?(@.userId == $userId)'
                        THEN jsonb_set(d.value, '{votes}',
                        (SELECT JSONB_AGG(
                        CASE when dd.value @? '$ ?(@.userId == $userId)' AND NOT dd.value -> 'option' @> '\"$option\"'::jsonb
                        THEN jsonb_set(jsonb_set(dd.value,
                        '{option}', '\"$option\"')
                        , '{date}', '\"$date\"')
                        ELSE dd.value
                        END)
                        FROM jsonb_array_elements(jsonb_path_query_array(d.value, '$.votes[*]')) dd)
                         )
                        ELSE jsonb_insert(d.value, '{votes, -1}', '{
                        \"userId\": $userId,
                        \"date\": \"$date\",
                        \"option\": \"$option\"
                        }', true)
                        END
                        ELSE d.value
                        END)
                        FROM jsonb_array_elements(data) d)
                        where id = ${pollModel.id}""",
            )
        }

Wenn wir drei Anführungszeichen schreiben, dann haben noch Unterstützung von der IDE.